controllers/listings.js

```diff
@@ -13,40 +13,92 @@ exports.getListings = asyncHandler(async (req, res) => {
   res.status(200).json(res.advancedResults);
 });
 
+/**
+ * @desc    Get all listings including soft deletes
+ * @route   GET /api/listings/all
+ * @access  Admin
+ */
+exports.getListingsAll = asyncHandler(async (req, res) => {
+  res.status(200).json(res.advancedResults);
+});
+
 /**
  * @desc    Get single listing by listing id
  * @route   GET /api/listings/:id/raw
  * @access  Public
+ * @desc    Get single listing by listing id including soft delete
+ * @route   GET /api/listings/:id/raw?all=true
+ * @access  Admin
  */
-exports.getListing = asyncHandler(async (req, res) => {
-  const rows = await db.one(
-    'SELECT * FROM listings WHERE listing_id = $1',
-    req.params.id
-  );
-  res.status(200).json({
-    success: true,
-    data: rows,
-  });
+exports.getListing = asyncHandler(async (req, res, next) => {
+
+  if (req.query.all === 'true') {
+    const rows = await db.one(
+      'SELECT * FROM listings WHERE listing_id = $1',
+      req.params.id
+    );
+
+    //if not admin and the listing has been deactivated, return a bad request
+    if ((!req.user || req.user.role !== 'admin') && rows.deleted_on !== 'null') {
+      return next(new ErrorResponse(`Listing does not exist`, 400));
+
+    } else {
+      res.status(200).json({
+        success: true,
+        data: rows,
+      })
+    }
+
+  } else {
+    const rows = await db.one(
+      'SELECT * FROM listingsview WHERE listing_id = $1',
```

Comment:
Following above comment:
```suggestion
      'SELECT * FROM listings WHERE listing_id = $1',
```